src/services/fetchAndTryLockCourtCase.ts

```diff
@@ -0,0 +1,41 @@
+import { DataSource } from "typeorm"
+import { isError } from "types/Result"
+import CourtCase from "./entities/CourtCase"
+import User from "./entities/User"
+import getCourtCase from "./getCourtCase"
+import tryToLockCourtCase from "./tryToLockCourtCase"
+
+type fetchAndTryLockCourtCaseResult = { courtCase?: CourtCase; error?: boolean }
+
+const fetchAndTryLockCourtCase = async (
+  currentUser: User,
+  courtCaseId: number,
+  dataSource: DataSource
+): Promise<fetchAndTryLockCourtCaseResult> => {
+  const ok = await tryToLockCourtCase(dataSource, courtCaseId, currentUser.username)
+  if (!ok) {
+    return {
+      error: true
+    }
+  }
+
+  const courtCase = await getCourtCase(dataSource, courtCaseId, currentUser.visibleForces)
+  if (!courtCase) {
```

Comment:
Is this a case of an error? `getCourtCase` would return null if the court case doesn't exist so I think we need to handle as not found

Reply:
If you look at L23 in the case details page (`index.tsx`), this gets handled as a `notFound` already 😄